checker.cs

```diff
@@ -1,38 +1,64 @@
 using System;
 using System.Diagnostics;
 
-class Checker
+namespace BatteryManagement
 {
-    static bool batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            Console.WriteLine("Temperature is out of range!");
-            return false;
-        } else if(soc < 20 || soc > 80) {
-            Console.WriteLine("State of Charge is out of range!");
-            return false;
-        } else if(chargeRate > 0.8) {
-            Console.WriteLine("Charge Rate is out of range!");
-            return false;
+    class Checker
+    {
+
+        static void IsBatteryOkay(BatteryMeasure measures, string Language)
+        {
+            CheckTemperature(measures.Temperature, Language);
+            CheckStateOfCharge(measures.StateOfCharge, Language);
+            CheckChargeRate(measures.ChargeRate, Language);
+            if(Language == "English")
```

Comment:
IsBatteryOkay method resist extensibility.... if we introduce new language , how does this method adopt new changes without modifying the code

Reply:
Passing Language as an input to the CallPrintMethods(). If we introduce new language,  addition of few lines of codes in the CallPrintMethods (with minimal code change), we can achieve the result.